### PR TITLE
MM-60722 - don't allow multiple '@' in email

### PR DESCRIPTION
**Pull request by sbishel**
#### Summary
This PR updates the Email Validation to no allow quoted strings.
It also adds tests to ensure emails using invalid characters also fails.

#### Ticket Link
Fixes https://mattermost.atlassian.net/browse/MM-60722

#### Release Note

```release-note
Add check for multiple '@' symbols in email address
```

### CODE REVIEW

File: server/public/model/utils.go [hunk 1]
```diff
@@ -631,6 +631,12 @@ func IsValidEmail(email string) bool {
 		return false
 	}
 
+	// mail.ParseAddress accepts quoted strings for the address
+	// which can lead to sending to the wrong email address
```
Comment (enzowritescode) [inline server/public/model/utils.go:635]: Can you `strings.ContainsAny(email, "\"") || strings.Count(email, "@") > 1` ? Also the comment will need updating

**Review by enzowritescode — CHANGES_REQUESTED** [review]

File: server/public/model/utils.go (hunk 1 above)
Comment (sbishel) [reply server/public/model/utils.go:635]: Its unnecessary...but sure I can add it.

**Review by enahum — APPROVED** [review]
Lgtm aside of the current discussion

**Review by enzowritescode — APPROVED** [review]

**Review by JulienTant — APPROVED** [review]

**Review by lindy65 — APPROVED** [review]
Thanks @sbishel - this is a little complex to test on the spinwick. I've QA-deferred for testing post merge

**Pull request merged**